### ISSUE TITLE
Simulation breaks after a while

### ISSUE THREAD

**Issue by emepetres** [gtorodelvalle/fiware-device-simulator#222]
Hi, I'm using the simulator with the orion global instance at FIWARE Lab. After a while (from 1 hour to 24 hours) the simulator always breaks, with two possible messages:

> {"time":"2016-10-28T10:39:31.674Z","lvl":"ERROR","msg":"error event: { error: { [Error: read ECONNRESET] code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' } }"}
> {"time":"2016-10-28T10:39:33.500Z","lvl":"INFO","msg":"end event"}

or

> {"time":"2016-10-29T01:14:51.170Z","lvl":"ERROR","msg":"error event: { error: \n   { name: 'TokenNotAvailable',\n     message: 'Authorization token could not be generated due to error
 (code: undefined, title: undefined, message: undefined)' } }"}
> {"time":"2016-10-29T01:17:52.429Z","lvl":"INFO","msg":"end event"}

At least the simulator should be able to reconnect and keep sending messages where it was (without starting again). Also I think that when the simulator breaks this way, an status different than 0 should be returned.

**Comment by gtorodelvalle** [comment]
Hi @emepetres , those messages are returned by the FIWARE Lab authorization (token generation) mechanism, so to say it, it is not on the simulators side but on the FIWARE Lab one, the simulator just logs them :) 

On the other hand, the FIWARE Device Simulator supports retrying when requesting tokens from FIWARE Lab infrastructure as stated in the documentation. See the `retry` property inside the top level `authorization` property of the simulation configuration files described at https://github.com/gtorodelvalle/fiware-device-simulator/blob/master/README.md Please, increase the number of retries to fit your needs... ;)

Last but not least, the simulator "does not break", it tries to contact a remote service the configured number of times and this remote entity is the one which is not online or at least not able to provide the information it is supposed to provide. After retrying it the configured number of times, the simulator logs the errors and exits normally.

I hope the `authorization.retry.times` and `authorization.retry.interval` properties of the simulation configuration files fits your need.
 
Thank you very much for your suggestions! ;)

**Comment by emepetres** [comment]
Hi Germán, thank you for letting me know, I totally missed that option! 

Anyway I've been testing it all weekend and I've noticed that, using the Orion Global Instance, after a few retries, the simulator seems to push data to Orion randomly, for example, if I send data every 30 minutes, it can stop sending data for two hours and then send a message. Finally I've just stopped using the global instance and start using our own instance instead, everything working neat!
